homework-submission/week4hw/src/components/TodoItem.js

```diff
@@ -0,0 +1,76 @@
+import React from "react";
+import Description from "./Description";
+import Button from "./Button";
+import "../styles/TodoItem.css";
+import { observer, inject } from "mobx-react";
+import { observable } from "mobx";
+
+@inject("todoListStore")
+@observer
+class TodoItem extends React.Component {
+  @observable isEditing = false;
+
+  editButtonHandler = () => {
+    this.isEditing = true;
+  };
+
+  cancelButtonHandler = () => {
+    this.isEditing = false;
+  };
+
+  updateInput = e => {
+    e.preventDefault();
+    this.props.todoListStore.updateInputHandler(e);
+    this.isEditing = false;
+  };
+  render() {
+    const { index, todo } = this.props;
+    const { onRemoveItemButton, onCheckboxChange } = this.props.todoListStore;
+
+    let class_name = "display_style";
+
+    if (this.isEditing) {
+      class_name = "editing_style";
+    }
+    return (
+      <div className="todo_item">
+        <div className="labels_checkbox">
+          <input
+            type="checkbox"
+            onChange={() => onCheckboxChange(index)}
+            checked={todo.done}
+          />
+          <label
+            htmlFor={todo.id}
+            style={{ textDecoration: todo.done ? "line-through" : "" }}
+          >
+            <Description
+              cancelButtonHandler={this.cancelButtonHandler}
+              updateInputHandler={this.updateInput}
+              isEditing={this.isEditing}
+              index={index}
+              description={todo.description}
+              deadline={todo.deadline}
+            />
+          </label>
+        </div>
+        <div className="buttons_container">
```

Comment:
what about editing the deadline?

Reply:
I didn't know it was required

Reply:
it's not, but how ever it's nice to see a complete challenge done🙂.

Reply:
I will try to add it tomorrow, i'm a little busy now :D

Reply:
don't worry take your time :)